src/anyon/anyon.jl

```diff
@@ -25,37 +26,105 @@ struct AnyonQPU <: AbstractQPU
     AnyonQPU(; host::String, user::String, access_token::String) = new(Client(host=host, user=user, access_token=access_token))
 end
 
-get_metadata(::AnyonQPU) = Dict{String,String}(
+const line_connectivity_label="linear"
+
+get_metadata(::AnyonQPU) = Dict{String,Union{String,Int}}(
     "manufacturer"  =>"Anyon Systems Inc.",
     "generation"    =>"Yukon",
     "serial_number" =>"ANYK202201",
+    "qubit_count"   =>6,
+    "connectivity_type"  =>line_connectivity_label
 )
 
 get_client(qpu_service::AnyonQPU)=qpu_service.client
 
+get_num_qubits(qpu::AnyonQPU)=get_metadata(qpu)["qubit_count"]
+
+get_connectivity(qpu::AnyonQPU)="1──2──3──4──5──6"
+
 function Base.show(io::IO, qpu::AnyonQPU)
     metadata=get_metadata(qpu)
 
     println(io, "Quantum Processing Unit:")
     println(io, "   manufacturer:  $(metadata["manufacturer"])")
     println(io, "   generation:    $(metadata["generation"])")
     println(io, "   serial_number: $(metadata["serial_number"])")
+    println(io, "   qubit_count:   $(metadata["qubit_count"])")
+    println(io, "   connectivity_type:  $(metadata["connectivity_type"])")
 end
 
-get_native_gate_types(::AnyonQPU)=[
-    Snowflake.PhaseShift,
-    Snowflake.Pi8,
-    Snowflake.Pi8Dagger,
-    Snowflake.SigmaX,
-    Snowflake.SigmaY,
-    Snowflake.SigmaZ,
-    Snowflake.X90,
-    Snowflake.XM90,
-    Snowflake.Y90,
-    Snowflake.YM90,
-    Snowflake.Z90,
-    Snowflake.ZM90,
-]
+
+function is_native_gate(qpu::AnyonQPU,gate::AbstractGate)::Bool
+    
+    set_of_native_gates=[
+        PhaseShift,
+        Pi8,
+        Pi8Dagger,
+        SigmaX,
+        SigmaY,
+        SigmaZ,
+        X90,
+        XM90,
+        Y90,
+        YM90,
+        Z90,
+        ZM90,
+        ControlZ,
+    ]
+
+    if typeof(gate)==ControlZ
+        @assert get_metadata(qpu)["connectivity_type"]==line_connectivity_label (
+            "Not implemented for connectivity type: $(get_metadata(qpu)["connectivity_type"])"
+        )
+            
+        targets=get_connected_qubits(gate)
+
+        return (abs(targets[1]-targets[2])==1)        
+    end
+        
+    return (typeof(gate) in set_of_native_gates)
+end
+
+function is_native_circuit(qpu::AnyonQPU,circuit::QuantumCircuit)::Bool
```

Comment:
It is unexpected that `is_native_circuit` throws an error if it returns a bool. I think it is better to have this return bool and throw an error in `run_job` if the circuit is not native.